File: totosuki/ABC/C/199.py
N = int(input())
S = list(input())
Q = int(input())

S1 = S[:N]
S2 = S[N:]

flag = False

for _ in range(Q):
    T, A, B = map(int, input().split())
    
    if T == 1:
        A, B = map(lambda x: x-1, [A, B])
        if A >= N: # Aが半分より大きかったら
            S2[A-N], S2[B-N] = S2[B-N], S2[A-N]
        elif B < N: # Bが半分より小さかったら
            S1[A], S1[B] = S1[B], S1[A]
        else: # それ以外（Aが半分より小さくて、Bが半分より大きい）
            S1[A], S2[B-N] = S2[B-N], S1[A]
    if T == 2:
        S1, S2 = S2, S1

print("".join(S1 + S2))

File: OCH_Tao/ABC/B/199.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const A = input[1].split(" ").map(x => parseInt(x, 10));
const B = input[2].split(" ").map(x => parseInt(x, 10));
console.log(Math.max(0, Math.min(...B) - Math.max(...A) + 1));